compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java

```diff
@@ -255,10 +255,42 @@ private BPackageSymbol definePackage(DataInputStream dataInStream, int pkgCpInde
         // Define service declarations
         defineSymbols(dataInStream, rethrow(this::defineServiceDeclarations));
 
+        populateReferencedFunctions();
+
         this.typeReader = null;
         return this.env.pkgSymbol;
     }
 
+    private void populateReferencedFunctions() {
```

Comment:
Please check the following sample, I do not get an error for not having an `xyz` method in `MyBar`.

foo module
```ballerina
public type FooObj object {
    public function xyz();
};

public type BarObj object {
    *FooObj;
    public function abc();
};
```

Module that imports `foo`
```ballerina
class MyBar {
    *foo:BarObj;

    public function abc() {} // should result in an error for missing `xyz`
}
```

Reply:
Added a bala test in 2a8045f1513eb0a447d358b96b77ad6ab656a715